rero/rero-ils
# users: patron in multiple organisations

* Replaces the string `current_patron` by `current_librarian` in the
  entire project.
* Adds a new `current_patrons` containing the list of the logged user
  patron accounts.
* Removes useless jinja detailed view configuration and templates.
* Fixes link redirection for the ill request.
* Cleans the patron request Web API.
* Fixes tests and permissions.
* Replaces `current_organisation.pid` by `current_librarian.organsiation_pid.
* Fixes tests and premissions.
* Gets the patron request rank given a patron instead of a barcode.
* Fixes translated string on circulation.
* Hides the profile menu given the right roles.
* Adds organisations information in the User API.
* Closes #1837.
* Closes #1460.
* Closes #1851.
* Closes #1804.
* Closes #1634.
* Closes #1490.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>
Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/us/2034?milestone=286652

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* https://github.com/rero/rero-ils-ui/pull/567


## For Reviewers
This huge PR contains 4 commits, you can use the followings links to make you review on each commit to make your job easier:

- https://github.com/rero/rero-ils/pull/1797/commits/94db328e99a35702f0e9d6099f43015a66457dba, is just a find and replace: `current_patron` -> `current_librarian`
- https://github.com/rero/rero-ils/pull/1797/commits/d5e1024f040cc4eed55fd5f3f1929da63823ada6, adds a new concept `current_patrons` which is the list of current patron account (one max per organisation) related to the current logged user.
- https://github.com/rero/rero-ils/pull/1797/commits/86de28071fe7a8ec3fa2df2bfc9e9df9bf41d42b, this is basically a refactoring commit by removing a useless `current_organisation` proxy.
- https://github.com/rero/rero-ils/pull/1797/commits/1af64a01fde0f863fd7e498795e6a3b7cb7e56e3, small fix for the UI interface.

## How to test?

- Simonetta is a patron in the two organisation 1 and 2
- Go to the admin view and try to create one of each type of resource (access rights)
- Check that the search list of each resource is filtered correctly.
- Try to attach a patron of an other organisation to your organisation and try to do a request, validate a request make checkin/checkout for this new patron. Check the patron profile of the newly created patron.
- Try to create a new librarian in your organisation which is already a librarian in an other org: you should not be able to do it.
- Try to generate a notification for a user with multiple patrons.
- Register a new user from the user interface and create a patron account.
- Change the patron information in the org1 et see if it is reported to the org2.
- Try to change the password for a user with two organisations.
- Check that the reset password instruction does not contains the library link.
- Check the patron request with items of two different organisations.
- Try to change the role of a patron with two organisations, the detailed view and the editor should contains the right roles.
- Try to change the "keep history" button in the patron profile and see that it is correction.
- Change the patron personal information in the patron profile and check that all the patrons informations are up to date in the professional views.
- Check the profile entry menu and ill request menu is correct for a librarian(hidden), patron and a user with a multiple patron accounts.
- etc.

## Code review check list
